### ISSUE TITLE
[BUG] ZYZ Euler extraction is ill-conditioned at beta->0: ~133 deg azimuth error at the geographic poles

### ISSUE THREAD

**Issue by christianhbye** [christianhbye/croissant#152]
**Describe the bug**

`rotations.rotmat_to_eulerZYZ` is ill-conditioned as `beta -> 0` or `beta -> pi`, because the matrix it is given is not orthogonal. At the geographic poles this produces a net sky-rotation error of tens to hundreds of degrees.

Two things combine:

1. `get_rot_mat` builds the topocentric→equatorial matrix by transforming three unit vectors through astropy's `SkyCoord.transform_to`. That path applies annual aberration, which is direction-dependent, so the three transformed vectors are not mutually orthogonal. Measured: `||R R^T - I||_max ~ 1e-4` (about 15-20 arcsec) at every latitude.

2. The ZYZ extraction divides by `sin(beta)` in effect, so that fixed ~1e-4 non-orthogonality is amplified by `1/sin(beta)`. Away from the poles it is invisible. At `beta -> 0` it dominates completely.

Since v5.3.0.dev2 the Earth simulation frame is CIRS at `times_jd[0]`, whose z-axis *is* the local zenith at latitude ±90, so `beta` collapses to ~20 arcsec and the amplification becomes catastrophic. Before that change the frame was J2000, whose pole sits ~428 arcsec off the zenith (precession since J2000), which accidentally conditioned the split — so this is a regression exposed by #147, not caused by it. #147 is correct; it removed the misalignment that had been masking this.

Note the existing `np.isclose(beta, 0.0)` gimbal-lock branch never fires here (atol 1e-8 rad ~ 0.002 arcsec). The damage is in the intermediate regime, so widening that tolerance is **not** the fix.

**To Reproduce**

```python
import numpy as np, jax
jax.config.update("jax_enable_x64", True)
import croissant as cro
from astropy.coordinates import AltAz, EarthLocation
from astropy.time import Time

def rz(t):
    c, s = np.cos(t), np.sin(t)
    return np.array([[c, -s, 0], [s, c, 0], [0, 0, 1]])

def ry(t):
    c, s = np.cos(t), np.sin(t)
    return np.array([[c, 0, s], [0, 1, 0], [-s, 0, c]])

def recon(m):
    a, b, g = [float(x) for x in cro.rotations.rotmat_to_eulerZYZ(m)]
    return rz(a) @ ry(b) @ rz(g), b

def angle_between(r1, r2):
    c = (np.trace(r1 @ r2.T) - 1) / 2
    return float(np.rad2deg(np.arccos(np.clip(c, -1, 1))))

t0 = Time("2022-07-17 00:00")
et = cro.rotations.jd_to_et(t0.tdb.jd)
for lat in [0.0, 45.0, 80.0, 89.0, 89.9, 90.0, -90.0]:
    loc = EarthLocation.from_geodetic(0.0, lat, height=0)
    topo = AltAz(obstime=t0, location=loc)
    r = np.asarray(cro.rotations._rot_mat_to_earth_sim_frame(topo, et=et))
    nonorth = np.max(np.abs(r @ r.T - np.eye(3)))
    u, _, vt = np.linalg.svd(r)
    r_true = u @ vt                      # nearest true rotation
    rec_raw, beta = recon(r)
    rec_ort, _ = recon(r_true)
    print(f"{lat:7.1f} {np.rad2deg(beta)*3600:10.1f}\" {nonorth:10.2e} "
          f"{angle_between(rec_raw, r_true):9.3f} "
          f"{angle_between(rec_ort, r_true):10.2e}")
```

Output on croissant v5.3.0.dev2 (`d972c5f`), jax 0.11.1. `err raw` and `err ortho` are the angular error of the Euler reconstruction against the nearest true rotation, in degrees — convention-independent, so it is meaningful at both `beta ~ 0` and `beta ~ pi`:

```
    lat         beta    |RRt-I|    err raw   err ortho
    0.0   324003.0"   1.12e-04     0.004    0.00e+00
   45.0   162001.6"   1.40e-04     0.010    1.71e-06
   80.0    35999.6"   1.10e-04     0.040    1.21e-06
   89.0     3599.1"   9.85e-05     0.372    1.21e-06
   89.9      359.5"   9.72e-05     3.705    0.00e+00
   90.0       19.4"   9.71e-05   133.355    0.00e+00
  -90.0   647979.7"   7.53e-05    37.771    0.00e+00
```

The error tracks `1/sin(beta)` exactly — each factor-10 reduction in `beta` gives a factor-10 growth in error — until it saturates at the pole. The south pole hits the `beta ~ pi` singularity and is affected too.

**Expected behavior**

The Euler reconstruction should recover the input rotation at all latitudes, as it does everywhere away from the poles. Orthogonalizing the matrix before extraction achieves that — the `err ortho` column above is the same code path fed `U @ Vt` from the SVD, and it is at machine precision at every latitude including both poles.

Suggested fix, in `get_rot_mat` (or immediately before `rotmat_to_eulerZYZ`):

```python
u, _, vt = np.linalg.svd(rmat)
rmat = u @ vt
```

This is the nearest orthogonal matrix in the Frobenius sense. It discards the ~15 arcsec of aberration-induced shear, which is not a rotation and cannot be represented by Euler angles anyway.

**Downstream symptom that led here**

Simulating the same sky and beam at the North Pole (lat 90, lon 0) and comparing against an independent reference simulator: agreement is 4.4 K mean under croissant 5.1.4 and 117.5 K under v5.3.0.dev2. The residual is flat in LST rather than time-varying, and a cyclic LST-lag scan finds it collapsing back to 1.65 K at a nonzero lag — i.e. the simulation is correct apart from a constant azimuth offset. Mid-latitude sites are unaffected and in fact improve under v5.3.0.dev2 (MARS, lat 79.4: 2.36 K -> 1.30 K, with the intra-day drift from #147 visibly removed).

**Desktop**
 - OS: linux
 - Version: v5.3.0.dev2 (`d972c5f`), jax 0.11.1, python 3.12
